src/fast_check/transform.rs

```diff
@@ -1154,6 +1277,119 @@ impl<'a> FastCheckTransformer<'a> {
     DiagnosticRange::new(self.specifier.clone(), range)
   }
 
+  fn maybe_transform_expr_if_leavable(
```

Comment:
Hmmm yeah tricky. I believe that the mutation is only necessary for if the expr is an `Arrow` or `Fn`, correct? It seems tempting to only support those as leavable at the top level so that we could re-use is_expr_leavable otherwise, but it also seems nice to do this recursrively as well. Perhaps we should just add a bit warning comment to both functions that we should keep them in sync?

Reply:
i'll add a comment